Vampire Mode

Super3222 opened the pull request:
### Vampire Mode

A gamemode focused on atrocious vampires draining the life out of the crew and advancing their objectives.

Depending on the amount of blood a vampire collects they will automatically unlock new sets of abilities/skills. Sometimes new, and sometimes strengthening old ones.

-----

### Gamemode

The design doc for the gamemode itself can be examined here: http://piratepad.net/HvCBjcZcAR

The gamemode will regularly begin with 3 vampires who's goals are to collect a certain amount of blood by the end of the shift, the vampire has a chance of getting a steal or kill objective (never together), and instead of an escape on the shuttle objective they have to survive until the end.

Vampires work alone; they have no reason to work together. Therefore, the gamemode does not have any mechanics that would help vampires work together. At the same time, there is no penalty for vampires that choose to fight against one another.

-----

### Progress

- [x] Gamemode

-----

### Credits

**Adam El-Tablawy** - Design Lead / Invented the Mode
**Super3222** - Coder, helped with design / Coded majority of the mode
**ShadowDeath6** - Coder
**Maya/Larissa** - Spriter
**Kmc2000** - Minor sprite work

The design doc also includes who did what. [DONE by] indicates it was coded by. If anything happens to the doc, please contact me. I may also be issuing another one to collect bug reports.

Big thanks to Maya, without them the project would've been stuck in development limbo. They were super helpful.

#### Changelog

:cl:
rscadd: INTRODUCING A NEW GAMEMODE LIKE YOU'VE NEVER SEEN BEFORE!
rscadd: The Court of Vampires has their eyes on Space Station 13! Look out for bloodsucking ghouls running amok on the station!
rscadd: Vampires, by draining blood you will be able to unlock unique abilities and strengthen your traits allowing you to become stronger.
rscadd: Crew, keep a look out for vicious vampires in the area. There's no telling what they might do!
/:cl:

matskuman5 commented:
cool

Time-Green commented:
Incredible work super, shadow and Maya.

God I can't wait for all this to get merged, so much. new and exciting content

ShadowDeath6 commented:
I didn't actually do anything don't fool yourselves

Alek2ander commented:
Cool.
Can we remove slings now?

oisin100 commented on code/game/objects/items/weapons/grenades/spawnergrenade.dm:
```diff
@@ -43,3 +43,8 @@
 	name = "Mister Scrubby"
 	spawner_type = /obj/item/weapon/soap/syndie
+
+/obj/item/weapon/grenade/spawnergrenade/clownbugs
```
UNDOCUMENTED FEATURE?!?!

Super3222 replied:
vampire mode was only a trojan horse for clownbugs

Time-Green commented:
Make sure they can't use mirrors

Time-Green commented:
Make sure they are invisible photographs and can't use mirrors

Super3222 commented:
> Make sure they are invisible photographs and can't use mirrors

Yup! We got appearing invisible in photographs down. Mirrors I'll jot down.

ShadowDeath6 commented:
mirrors might actually hurt them p bad because they can't change their appearance at all

Mrcreepssss commented:
maybe mirror should randomize look instead? seeing as they could still change their appearance technically, it'd just be a mess without mirror.

Super3222 commented on code/_onclick/click.dm:
```diff
@@ -405,7 +408,7 @@
 		if(istype(glasses, /obj/item/clothing/glasses/hud/security) || istype(CIH,/obj/item/organ/cyberimp/eyes/hud/security))
 			var/allowed_access = checkHUDaccess(1)
 			if(allowed_access)
-				security_scan_status(src, A, allowed_access)
```
uhuh

Super3222 commented on code/_onclick/click.dm:
```diff
@@ -415,15 +418,15 @@
 		if(wear_id)
 			var/obj/item/weapon/card/id/I = get_idcard()
-			if((1 in I.access))
+			if((1 in I))
```
merg conflicts

Super3222 commented:
This mode was completed a long time ago

Time-Green commented:
Happy 2 months anniversairy!

X-TheDark commented on code/game/gamemodes/vampire/vamp_powers/vamp_400.dm:
```diff
@@ -0,0 +1,74 @@
+				playsound(L, 'sound/machines/warning-buzzer.ogg', 50, 1)
+
+	for(var/obj/structure/window/reinforced/R in view(12, H))
+		R.take_damage(100)
```
Why are reinforced glass windows damaged in 12 tile radius instead of 7 as normal windows?

Super3222 replied:
I wanted to make it as loud as possible, and more powerful than a shadowling or horrorform shriek (w/o horrorforms intense stun). But if you want it to be 7 instead of 12 just to keep things in close range I can do that.

X-TheDark replied:
It's not the number that bothers me, it's that it's different for several types of stuff. 7 for one, 12 for some others.

Either make all of them 7 or make all of them 12.

X-TheDark commented on code/game/gamemodes/vampire/vamp_powers/vamp_400.dm:
```diff
@@ -0,0 +1,74 @@
+		if(prob(60))
+			G.Break()
+	for(var/obj/machinery/computer/C in view(12, H))
+		C.take_damage(100)
```
Same question, why 12 instead of 7?

X-TheDark commented on code/game/gamemodes/vampire/vamp_powers/vamp_600.dm:
```diff
@@ -0,0 +1,94 @@
+			continue
+		if(!chosenone)
+			O << "<span class='warning'>[H] scratches at your floating waist, reeling you in!</span>"
+			chosenone = O
```
Minor point - at some point in the future, implement Aghosted admin checking. Don't want vampires sucking off admin ghosts. Not a showstopper.

X-TheDark commented on code/game/gamemodes/vampire/vamp_powers/vamp_800.dm:
```diff
@@ -0,0 +1,37 @@
+		return
+
+	if(!(istype(charm, /mob/living/simple_animal/hostile)))
+		var/mob/living/simple_animal/hostile/mimic/copy/C
```
Nothing is assigned to this.

X-TheDark commented on code/game/gamemodes/vampire/vamp_powers/vamp_800.dm:
```diff
@@ -0,0 +1,37 @@
+
+	if(!(istype(charm, /mob/living/simple_animal/hostile)))
+		var/mob/living/simple_animal/hostile/mimic/copy/C
+		C.check_obj = FALSE
```
C doesn't exist at this point, you need to move `new()` up.

X-TheDark commented on code/game/gamemodes/vampire/vamp_powers/vamp_800.dm:
```diff
@@ -0,0 +1,37 @@
+		C = new(get_turf(H), charm, H, destroy_original = FALSE, eyes = FALSE)
+		C.faction += "Vampire"
+	else
+		charm.faction += "Vampire"
```
`faction |= "Vampire" for both, instead of `+`

X-TheDark commented on code/game/gamemodes/vampire/vampire.dm:
```diff
@@ -0,0 +1,140 @@
+	protected_jobs = list("Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Chaplain")
+	required_players = 1 // 15
+	required_enemies = 0 // 3
+	recommended_enemies = 0 // 1
```
Notice for after testmerge - change these to the proper values.

X-TheDark commented on code/game/gamemodes/vampire/vampire.dm:
```diff
@@ -0,0 +1,140 @@
+	return 1
+
+/datum/game_mode/vampire/proc/greet_vampire(datum/mind/vampire)
+	vampire.current << "<B><span class='announcevampire'>You are a Vampire.</font></B>"
```
Are you sure it's `</font>` instead of `</span>`?

Super3222 replied:
Probably, yeah.

Cruix commented on code/game/gamemodes/objective.dm:
```diff
@@ -931,8 +932,57 @@ var/global/list/possible_items_special = list()
+			if(800)
+				pass = owner.vampire.thousand_unlocked
+			if(1000)
+				pass = owner.vampire.eighthundred_unlocked
```
These are backwards

Super3222 replied:
They make pass TRUE or re-assign it as FALSE though. How're they backwards?

X-TheDark replied:
Cruix meant this
    if(800)
        pass = owner.vampire.>>thousand_unlocked<<

Cruix commented on code/game/gamemodes/objective.dm:
```diff
@@ -931,8 +932,57 @@ var/global/list/possible_items_special = list()
+/datum/objective/charmIan/check_completion()
+	var/pass = FALSE
+	for(var/mob/living/simple_animal/pet/dog/corgi/Ian/I in living_mob_list)
+		if(I.faction == "Vampire")
```
Factions are lists, so this will always be false

Cruix commented on code/game/gamemodes/vampire/vamp_misc/vamp_structures.dm:
```diff
@@ -0,0 +1,45 @@
+		if(qdeleted(src))
+			vdatum.coffin = null
+			vdatum = null
+			STOP_PROCESSING(SSobj, src)
```
All of this should be in Destroy(), not process()

X-TheDark commented on code/game/gamemodes/vampire/vdatum.dm:
```diff
@@ -0,0 +1,225 @@
+	var/eighthundred_unlocked
+	var/thousand_unlocked
+
+/datum/vampire/proc/add_blood(amount, amthigh, amtlow)
```
I'm curious because I don't see the other two parameters currently used:
Anything specific planned for the parameters that are used for random amount of blood giving?

Super3222 replied:
When I was designing the vampire datum I wanted to leave potential for the future so in case someone (or even Adam) comes up with a crazy idea it could have a system for it.

Cruix commented on code/game/gamemodes/vampire/vamp_powers/basic.dm:
```diff
@@ -0,0 +1,246 @@
+
+
+/obj/effect/proc_holder/vampire/bite/proc/check_status(mob/living/L, datum/vampire/V, mob/living/carbon/human/T, rate)
+	if(L.weakened || L.stunned || L.sleeping || L.stat == DEAD || L.stat == UNCONSCIOUS || get_dist(L, T) > 1 || !L.pulling)
```
use L.incapacitated() and L.stat == CONCIOUS

Cruix commented on code/game/gamemodes/vampire/vamp_powers/basic.dm:
```diff
@@ -0,0 +1,246 @@
+		L << "<span class='alertvampire'>You've been interrupted!</span>"
+		return 0
+	if(!T.blood_volume || T.blood_volume < rate)
+		L << "<span class='noticevampire'>[T] has ran out of blood.</span>"
```
run out of blood

X-TheDark commented on code/game/gamemodes/vampire/vdatum.dm:
```diff
@@ -0,0 +1,225 @@
+	vampire.dna.species.specflags |= NOBLOOD
+	vampire.dna.species.specflags |= NOBREATH
+
+	vampire.faction += "Vampire"
```
`faction |= "Vampire"` to prevent duplicate entries in the list.

Cruix commented on code/game/gamemodes/vampire/vamp_powers/vamp_100.dm:
```diff
@@ -0,0 +1,91 @@
+
+/obj/effect/proc_holder/vampire/hypno/proc/sleeptarget(mob/living, mob/user) // in the future, make it check for a range so that the target can get away? or make it check for a garlic necklace.
+	if (living)
+		if(get_dist(user, living) > 7) // 7 range
```
Should this be < 7 instead of > 7?

AdamElTablawy replied:
IIRC this means a vampire can hit anyone outside of 7 tiles, which is hysterical but not what anyone wants

X-TheDark commented on code/game/gamemodes/vampire/vdatum.dm:
```diff
@@ -0,0 +1,225 @@
+// under get_limit() we are using the vampire's current unlock requirements as a limit towards \
+ how much the vampire can do X (whatever you want to use the limit for).
+// For instance, this is used as a cap towards how much blood a vampire can regenerate back inside of \
+ their coffin.
```
According to documentation, BYOND comments don't work this way, you can't add lines via an endline `\`
Prepend both 212 and 214 with `//`

Super3222 replied:
This is what it looks like in the code (I'm still reverting it though): https://image.prntscr.com/image/9eraY7QJRP_dB2q300doCQ.png

Cruix commented on code/game/gamemodes/vampire/vamp_powers/vamp_300.dm:
```diff
@@ -0,0 +1,109 @@
+	cooldown_immediately = FALSE
+	cooldownlen = 0
+	var/mob/living/simple_animal/hostile/bat/vampire/vbat = null
+	var/transformed = FALSE
```
This verb is never set to TRUE, so I would guess that you could transform into a bat if you are already one

Cruix replied:
This ability should reuse code from the wizard shapeshifting spell instead of copypasting it, since it is doing exactly the same thing

Cruix commented on code/game/gamemodes/vampire/vamp_powers/vamp_300.dm:
```diff
@@ -0,0 +1,109 @@
+
+	for(var/obj/item/device/radio/R in view(5,H))
+		if(get_dist(H, get_turf(H)) > 5)
+			continue
```
This will never be true

Cruix commented on code/game/gamemodes/vampire/vamp_powers/vamp_300.dm:
```diff
@@ -0,0 +1,109 @@
+
+	for(var/mob/living in view(5,H))
+		if(get_dist(H, get_turf(H)) > 5)
+			continue
```
This will never be true

X-TheDark commented on code/game/gamemodes/vampire/vdatum.dm:
```diff
@@ -0,0 +1,225 @@
+	if(bloodcount >= 400) limit = 400
+	if(bloodcount >= 600) limit = 600
+	if(bloodcount >= 800) limit = 800
+	if(bloodcount >= 1000) limit = 1000
```
Make this dependent on your current unlock stages `hundred_unlock`/etc, otherwise this is useless for regeneration purposes (where it's currently used), as it always returns a number less than your current bloodcount.

Cruix commented on code/game/gamemodes/vampire/vamp_powers/vamp_300.dm:
```diff
@@ -0,0 +1,109 @@
+		if(get_dist(H, get_turf(H)) > 5)
+			continue
+		var/turf/T = get_turf(living)
+		if(T && T.flags & NOJAUNT) // aka blessed.
```
T.flags & NOJAUNT should be in parenthesis

Cruix commented on code/game/gamemodes/vampire/vamp_powers/vamp_300.dm:
```diff
@@ -0,0 +1,109 @@
+		if(T && T.flags & NOJAUNT) // aka blessed.
+			continue
+		if(ishuman(living))
+			for(var/obj/item/device/radio/radio in living.contents)
```
use GetAllContents()

Cruix commented on code/game/gamemodes/vampire/vamp_powers/vamp_400.dm:
```diff
@@ -0,0 +1,74 @@
+		"<span class='warning'>[H] releases a horrifying screech!</span>")
+
+	for(var/turf/T in view(7,H))
+		if(istype(T, /obj/structure/window))
```
This will never be true, because windows are not turfs

X-TheDark commented on code/modules/client/verbs/suicide.dm:
```diff
@@ -10,6 +10,13 @@
+		if(H.mind.vampire)
+			if(H.mind.vampire.isDraining)
+				src << "<span class='vampirenotice'>You can't escape from me....</span>"
+				H << "<span class='vampirealert'>[src] has attempted to suicide, but was prevented!</span>"
```
Might want to ask around admin staff if this should be logged.

Cruix replied:
attempted to commit suicide

X-TheDark commented on code/modules/mob/living/simple_animal/hostile/retaliate/bat.dm:
```diff
@@ -33,4 +33,26 @@
+	qdel(src)
+	S.vbat = null
+	for(var/obj/effect/proc_holder/vampire/battrans/B in vamp.abilities)
+		B.switchonCD(150)
```
Move `qdel` to the end for safety. It's bad practice to try to access variables of something you're deleting, even if it's queued.

X-TheDark replied:
Move `qdel` to the end for safety. It's bad practice to try to access variables of something you're deleting, even if it's queued.

X-TheDark commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -450,7 +453,7 @@
 	description = "A vegetalizing toxin produced by slimes."
 	color = "#5EFF3B" //RGB: 94, 255, 59
-	race = /datum/species/plant/pod
+	race = /datum/species/plant
```
Hmmm?

Super3222 replied:
It seems to have been fuckery from a merge conflict.

Cruix commented on code/modules/mob/living/simple_animal/hostile/retaliate/bat.dm:
```diff
@@ -33,4 +33,26 @@
+
+	src << "<span class='noticevampire'>You return to a humanoid form.</span>"
+	vamp.forceMove(loc)
+	vamp.status_flags &= GODMODE
```
&= ~GODMODE

X-TheDark replied:
Pretty sure it's `status_flags -= GODMODE`, actually, as status_flags is a list now.

AdamElTablawy replied:
Why does the bat get godmode

X-TheDark replied:
Not the bat, it's the vampire inside that gets GODMODE, presumably, so the vampire doesn't get double-damage from explosions (even small ones).

Super3222 replied:
It's to prevent their human mob from suffocating from being placed inside of the bat. Shapeshift (wizard spell) does something similar to this.

Cruix commented on code/modules/paperwork/photography.dm:
```diff
@@ -248,6 +248,8 @@
 /obj/item/device/camera/proc/camera_get_mobs(turf/the_turf)
 	var/mob_detail
 	for(var/mob/M in the_turf)
+		if(M.mind.vampire)
```
M.mind might be null

Cruix commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -1471,3 +1474,21 @@ datum/reagent/romerol
+/datum/reagent/vampire
+	name = "unknown serum"
+	id = "vamp"
+	description = "A miracle chemical. One that's never been seen before."
```
One that has never been seen before

ShadowDeath6 replied:
if it's never been seen before wouldn't it be better to call it an unidentified reagent or something

Cruix commented on code/game/gamemodes/vampire/vamp_powers/vamp_300.dm:
```diff
@@ -0,0 +1,109 @@
+			vbat.S = src
+		H.forceMove(vbat)
+		H.status_flags |= GODMODE
+		vbat.verbs += /mob/living/simple_animal/hostile/bat/vampire/verb/humanform
```
This is a verb, so all vampire bats will have this by default

X-TheDark commented on code/_onclick/click.dm:
```diff
@@ -415,7 +418,7 @@
 			var/allowed_access = checkHUDaccess(1)
 			if(allowed_access)
-				security_scan_crime(src, A, allowed_access) // check hud.dm
+				security_scan_status(src, A, allowed_access) // check hud.dm
```
What does this change do, if you don't mind me asking?

Super3222 replied:
Merge conflict covfefe. I was probably trying to keep my changes to vampire click abilities.

ShadowDeath6 commented:
now you just need to add the stone mask item so people in lavaland can become vampires

Cruix commented:
Vampires do not show up in check-antags
There is no click delay on biting
Cannot add or remove vampires in the traitor panel
Vampires cannot see their ability cooldowns
Vampires to not take damage in the chapel
Selecting hypnotize for fast-casting puts it on cooldown
During the testmerge, one of the vampires lost all his abilities for no apparent reason

Complaints specifically from Adam:
Vampires die way to fast in light
I cannot pull my own coffin
I cannot select the ghost I want to eat
Sucking all the blood out of someone does not kill them if they ate tons of oxyloss healing chemicals

AdamElTablawy commented:
ok, the problems with this mode:

bite has no delay before you bite, meaning you can grab someone and immediately chomp them for -70 blood, and can constantly chain this to kill people in less than two seconds with skillful grab + bite spam

70 blood drain is probably too much

you can bite through hardsuits 

bite doesn't seem to give a notification to anyone nearby

selecting hypnotize puts hypnotize on cooldown, and it has a delay of a while before it actually works, does not have a progress bar, and doesn't need you to be near the guy when using

you cannot swap between vampiric gaze and hypnotize because you can't deselect an ability while it's on cooldown

there's no 'vampire' tab where you can view the cooldowns of abilities

there's no notification for the new abilities you gain, just they suddenly pop up

there's no brief descriptor when you spawn as to what you can do/what you're weak to

you die in the hallways almost immediately, scrap the 'weak to bright lights' shit

you cannot move your own designated coffin

it takes like 15 people to get 700 blood

there's no blood counter so you can't see how much you have or how much an ability will consume

Super3222 commented:
# Testmerge Patch

- Added a click delay to bite
- There is now a 3 second delay for bites before you start draining them.
- Vampires are now in check antagonist
- Vampires are now in traitor panel (where you can make someone a vampire or remove it from them)
- Vampires take burn damage (5 every tick) from being in the chapel
- Vampires taking damage from light idea scrapped
- Scrapped the 'coffin can only be summoned by vampire, and not pulled around' idea. Now only the vampire can drag it around. Anyone else's grab will get cancelled out
- While draining blood from someone, if the target runs out of blood their reagents will b drained
- Bat Swarm fixed (shadow you spawn something on a turf of a thing that wasn't been created yet!)
- Draining blood is loud, like it always should have been.
- Vampires can no longer bite through hardsuits
- Hypnotize mechanics are explained better. Ie. after using it your target has to be within rang for four seconds.
- Bat Swarm now randomizes 6-8 bats
- Added CoD to sixhundred unlock. For some reason it wasn't added.
- Drainrate:Drainpayoff ratio deduced from 55:10 -> 30:10
- Hypnotize no longer immediately goes on cooldown
- Added a blood hud so you can track how much blood you have
- Added a vampire tab where you can keep track of cooldowns.

## Images:

![](https://image.prntscr.com/image/ziHOQ9FaQoOXo6e_ymVirw.png)
![](https://image.prntscr.com/image/wo55VxzHS86FAfWGjKQ8tg.png)
![](https://image.prntscr.com/image/t77pfasNT2yOPaab8ie6Cg.png)
![](https://image.prntscr.com/image/IEbGtgASRt60fdZH1xh07A.png)
![](https://image.prntscr.com/image/YEXKySkWQlGe7tQEulEA5Q.png)
![](https://image.prntscr.com/image/5p0T5AYvQH24oGMEZiCC_g.png)
![](https://image.prntscr.com/image/aK6M8yFpRLmdBxhRTV0RiA.png)

### Other stuff:

- For ghost rend to work you need a ghost adjacent to you.
- I have no idea what caused that vampire to lose all of their abilities. Hopefully it comes up again in the future.

ShadowDeath6 commented on code/game/gamemodes/vampire/vamp_misc/vamp_structures.dm:
```diff
@@ -19,6 +18,10 @@
+	if(pulledby)
+		if(vdatum.vampire != pulledby)
+			pulledby.stop_pulling()
+			pulledby << "<span class='notice'>You oddly lose grip of [src].</span>"
```
oddly?

ShadowDeath6 commented on code/game/gamemodes/vampire/vamp_powers/basic.dm:
```diff
@@ -25,45 +26,66 @@
-		H << "<span class='alertvampire'>This one has a strange odor.</span>"
+		drainrate = 30
+		drainpayoff = 1
+		H << "<span class='alertvampire'>This one has a strange odor. It won't do much for you because they're dead.</span>"
```
remove the first sentence here

ShadowDeath6 commented on code/game/gamemodes/vampire/vamp_misc/vamp_structures.dm:
```diff
@@ -0,0 +1,46 @@
+	if(pulledby)
+		if(vdatum.vampire != pulledby)
+			pulledby.stop_pulling()
+			pulledby << "<span class='notice'>You oddly lose grip of [src].</span>"
```
say somehow or something, not oddly

ShadowDeath6 commented on code/game/gamemodes/vampire/vamp_powers/basic.dm:
```diff
@@ -0,0 +1,267 @@
+	if(target.stat == DEAD)
+		drainrate = 30
+		drainpayoff = 1
+		H << "<span class='alertvampire'>This one has a strange odor. It won't do much for you because they're dead.</span>"
```
remove the first sentence here if you're saying they're dead outright

ShadowDeath6 commented:
@Super3222 in the screenshot it says "The Jalen Beck is now under your control", is that fixed

Super3222 commented:
Yes.

nichlas0010 commented:
![image](https://user-images.githubusercontent.com/20558591/29814843-032ac344-8caf-11e7-81ee-a985bd66cddc.png)
 👌

ShadowDeath6 commented:
what's the status on this pr anyway

Super3222 commented:
ready for merge

Cruix commented:
Please remove the change to the gamemode file so I don't have to change it back to extended every time I pull from this branch.

I pulled this, made myself a vampire in the traitor panel, and then tried using every ability, as I expected you had already done while testing it yourself.
* You cannot un-vampire a person who did not start the round as a vampire, because making them a vampire does not add them to the vampire list. You can, however, vampire a person more than once, in which case they will get all of the vampire abilities multiple times.
* When you get to the blood level that lets you see ghosts, you will no longer be able to see them if you update your vision (such as by putting on or removing glasses). While you can see them, however, you can also see pipes under the floor. Trying to fix this by calling the level-up proc again on my vampire datum worked, but also gave me more copies of the abilities earned at that level.
* Blood tracking does not work at all. This is probably because the blood_DNA variable is for the DNA of any blood on an object that was put there by hitting someone with it, and as such, neither humans nor blood drips have it set, because they are never used to hit things. Also, blood trails are not of the /obj/effect/decal/cleanable/blood type, so you cannot even try to use the ability on them.
* Cloak of darkness runtimes whenever it is used and does nothing.
* When you use hypnotize, you twirl your finger when the ability is selected but not when you use it. I am not sure if this is intentional. You can also try to put things to sleep that cannot sleep, such as cyborgs and simple animals.
* Charm creature, when used on a friendly mob (I used a corgi), does not delete the mob and spawns a crate mimic. It really should not be deleting mobs in the first place. Also, it should not work on player-controlled mobs at all.
* You do not get a message when you lose your grip on a vampire's coffin.
* Whenever your shriek nearby a grille you hear "Agonizing Shriek whips back and forth repelling the scream!"

Super3222 commented:
I checked the code for cloak of darkness (in mutations.dm) and its defines, and nothing looks off. Can I get more information on that runtime.

Super3222 commented:
how the new blood track works
![](https://image.prntscr.com/image/WzcZIKK3Q0WDNLwkWBSIQQ.png)

ShadowDeath6 commented on code/game/gamemodes/vampire/vamp_powers/basic.dm:
```diff
@@ -0,0 +1,272 @@
+		for(var/mob/living/carbon/human/caughthuman in humanscaught)
+			H << "<span class='noticevampire'>[caughthuman] is in the [get_area(caughthuman)], just [dir2text(get_dir(get_turf(caughthuman), get_turf(H)))] from your location.</span>"
+	else
+		H << "<span class='alertvampire'>You couldn't find who the blood belonged too.</span>"
```
to

ShadowDeath6 commented on code/game/gamemodes/vampire/vamp_powers/basic.dm:
```diff
@@ -0,0 +1,272 @@
+	if(humanscaught.len)
+		H << "<span class='alertvampire'>You search for who the blood belongs too.</span>"
+		for(var/mob/living/carbon/human/caughthuman in humanscaught)
+			H << "<span class='noticevampire'>[caughthuman] is in the [get_area(caughthuman)], just [dir2text(get_dir(get_turf(caughthuman), get_turf(H)))] from your location.</span>"
```
I don't think you need the word "just" in this sentence

ShadowDeath6 commented on code/game/gamemodes/vampire/vamp_powers/basic.dm:
```diff
@@ -0,0 +1,272 @@
+		locatedblood_DNA = null
+
+	if(humanscaught.len)
+		H << "<span class='alertvampire'>You search for who the blood belongs too.</span>"
```
to

ShadowDeath6 commented on code/game/gamemodes/vampire/vamp_powers/vamp_100.dm:
```diff
@@ -0,0 +1,98 @@
+	V.chosen_click_attack = src
+	H << "<span class='vampirealert'>[src] is now active. (Use your middle mouse button on anyone \
+		to use.)"
+	H.visible_message("<span class='warning'>[H] twirls their finger in a circlular motion.</span>",\
```
I don't think this is a good idea for a toggleable thing, it should only display a message when you use it. Making it show a message really fucks the stealth aspect and leads to dumb stuff where you hide in a locker and do it to avoid the message

ShadowDeath6 commented on code/game/gamemodes/vampire/vamp_powers/vamp_1000.dm:
```diff
@@ -0,0 +1,56 @@
+	return 1
+
+/obj/effect/proc_holder/vampire/setsummonturf
+	name = "Set Summon Turf"
```
I'd say "location" rather than turf, it's preferential though

Slinky5890 commented:
Siiigh, and we were so close

ShadowDeath6 commented:
let him rest

AdamElTablawy commented:
GIVE ME THE SUCC.

Slinky5890 commented:
Monthly reminder we almost had this

Cruix commented:
* The default gamemode is still Vampire when it should be extended.
* You still cannot un-vampire a person who did not start the round as a vampire, and you still can vampire them more than once, in which case they will get all of the vampire abilities multiple times.
* While having ghost vision no longer lets you see pipes, it does make your darkness vision no longer work.
* Cloak of darkness still runtimes whenever it is used and does nothing.
* When you use hypnotize, you still twirl your finger when the ability is selected but not when you use it.
* Charm creature, when used on a friendly mob will teleport the mimic to you.
* Blood tracking now crashes the server.

Please test every ability once as a bare minimum of testing before reopening this.

AdamElTablawy commented:
@Super3222 LI V E

Super3222 commented:
![image](https://user-images.githubusercontent.com/16735500/34343628-a30a6084-e9a4-11e7-9051-9507ef2ab5ae.png)
